bayes_opt/acquisition.py

```diff
@@ -0,0 +1,854 @@
+        dummies = []
+        for dummy in self.dummies:
+            close = np.isclose(dummy, target_space.params, rtol=self.rtol, atol=self.atol)
+            if not close.all(axis=1).any():
+                dummies.append(dummy)
+        self.dummies = dummies
+        
+    def suggest(self, gp: GaussianProcessRegressor, target_space: TargetSpace, n_random=10_000, n_l_bfgs_b=10, fit_gp:bool=True) -> np.ndarray:
+        """Suggest a promising point to probe next.
+
+        Parameters
+        ----------
+        gp : GaussianProcessRegressor
+            A fitted Gaussian Process.
+
+        target_space : TargetSpace
+            The target space to probe.
+
+        n_random : int, default 10_000
+            Number of random samples to use.
+
+        n_l_bfgs_b : int, default 10
+            Number of starting points for the L-BFGS-B optimizer.
+
+        fit_gp : bool, default True
+            Whether to fit the Gaussian Process to the target space.
+            Set to False if the GP is already fitted.
+
+        Returns
+        -------
+        np.ndarray
+            Suggested point to probe next.
+        """
+        if len(target_space) == 0:
+            msg = (
+                "Cannot suggest a point without previous samples. Use "
+                " target_space.random_sample() to generate a point and "
+                " target_space.probe(*) to evaluate it."
+            )
+            raise TargetSpaceEmptyError(msg)
+
+        if target_space.constraint is not None:
+            msg = (
+                f"Received constraints, but acquisition function {type(self)} "
+                + "does not support constrained optimization."
+            )
+            raise ConstraintNotSupportedError(msg)
+    
+        # Check if any dummies have been evaluated and remove them
+        self._remove_expired_dummies(target_space)
+
+        # Create a copy of the target space
+        dummy_target_space = self._copy_target_space(target_space)
+
+        # Choose the dummy target value
+        if isinstance(self.strategy, Number):
+            dummy_target = self.strategy
+        elif self.strategy == 'min':
+            dummy_target = target_space.target.min()
+        elif self.strategy == 'mean':
+            dummy_target = target_space.target.mean()
+        else:
+            assert self.strategy == 'max'
+            dummy_target = target_space.target.max()
+
+        # Register the dummies to the dummy target space
+        for dummy in self.dummies:
+            dummy_target_space.register(dummy, dummy_target)
+
+        # Fit the GP to the dummy target space and suggest a point
+        self._fit_gp(gp=gp, target_space=dummy_target_space)
+        x_max = self.base_acquisition.suggest(gp, dummy_target_space, n_random=n_random, n_l_bfgs_b=n_l_bfgs_b, fit_gp=False)
+
+        # Register the suggested point as a dummy
+        self.dummies.append(x_max)
+
+        return x_max
+
+
+class GPHedge(AcquisitionFunction):
```

Comment:
I didn't want to go overboard and add too many acquisition functions. `GPHedge` and the `ConstantLiar` seemed appropriate. EIpu (#393) is additionally showcased in the notebook.